feature-settings-impl/src/main/java/io/novafoundation/nova/feature_settings_impl/presentation/networkManagement/networkList/preConfiguredNetworks/PreConfiguredNetworksViewModel.kt

```diff
@@ -58,18 +71,52 @@ class PreConfiguredNetworksViewModel(
     }
 
     fun networkClicked(chainId: String) {
-        showMessage("Not implemented yet")
+        launch {
+            progressDialogMixin.startProgress(R.string.loading_network_info) {
+                interactor.getPreConfiguredNetwork(chainId)
+                    .onSuccess {
+                        openAddChainScreen(it)
+                    }
+                    .onFailure { showError(resourceManager.getString(R.string.common_something_went_wrong_title)) }
+            }
+        }
+    }
+
+    private fun openAddChainScreen(chain: Chain) {
+        val (mode, chainId) = when (chain.networkType()) {
+            NetworkType.SUBSTRATE -> AddNetworkPayload.Mode.SUBSTRATE to null
+            NetworkType.EVM -> AddNetworkPayload.Mode.EVM to chain.evmChainIdOrNull()
+        }
+
+        val node = chain.nodes.nodes.firstOrNull()
+        val asset = chain.assets.firstOrNull()
+        val explorer = chain.explorers.firstOrNull()
+
+        val payload = AddNetworkPayload(
+            mode,
+            AddNetworkPayload.NetworkData(
+                iconUrl = chain.icon,
+                rpcNodeUrl = node?.unformattedUrl,
+                networkName = chain.name,
+                tokenName = asset?.symbol?.value,
+                evmChainId = chainId,
+                blockExplorerUrl = explorer?.normalizedUrl(),
+                coingeckoLink = asset?.priceId?.let { coinGeckoLinkParser.format(it) }
+            )
```

Comment:
We should allow for the whole chain information to be passed to the add screen, otherwise we wont be able to to enable staking or governance for pre-configured networks for example